paper_framework_sources/p2079_system_execution_context.md

```diff
@@ -178,14 +181,163 @@ public:
 `schedule` on a `system_scheduler` returns some implementation-defined `sender` type.
 
 This sender satisfies the following properties:
-  - Implements the `get_completion_scheduler` query for the value and done channel where it returns a type that is a pair
-    of an object that compares equal to itself, and a representation of delegatee scheduler that may be obtained from
+
+  - Implements the `get_completion_scheduler` query for the value and done channel where it returns a type that is logically
+    a pair of an object that compares equal to itself, and a representation of delegatee scheduler that may be obtained from
     receivers connected with the sender.
   - If connected with a `receiver` that supports the `get_stop_token` query, if that `stop_token` is stopped, operations
     on which `start` has been called, but are not yet running (and are hence not yet guaranteed to make progress) **must**
     complete with `set_done` as soon as is practical.
   - `connect`ing the `sender` and calling `start()` on the resulting operation state are non-blocking operations.
 
+# Design discussion and decisions
+## To drive or not to drive
```

Comment:
I spent a bit of time thinking about the drive operation since most of Intel’s threading runtimes expect the main thread to participate in the thread pool, even as perhaps the only available thread.  For OpenMP, the calling thread (when _not_ using offload) always participates, since you need to start parallel work from a thread team, which the caller is always part of. And for TBB, the blocking algorithms, such as parallel_for, include the calling thread in the pool. Even so, I think we have no issue with the system context. We can get the main thread to participate in the system context’s thread pool during a customized sync_wait.  And then for async_scope, we want eager execution anyway, so it doesn’t make sense to include the main thread – or again – it could participate when calling sync_wait on the completion sender. 

As for the standard drive or implementation defined drive -- I agree that implementation defined makes sense. In fact, if calling drive were required but its effectively a no-op on some platforms, it could just be a bug waiting to happen in people's code.